docs/build/guides/conventions/analyzing-smart-contract-cost.mdx

```diff
@@ -0,0 +1,449 @@
+- Examples: Oracle data, claimable balances, offers.
+
+#### Cost
+
+- Cheaper than persistent storage due to the limited lifespan of data. (Cheapest cost)
+
+#### Lifetime
+
+- Data exists for a predefined period and is then removed
+
+### [`env.storage().instance()`](https://docs.rs/soroban-sdk/latest/soroban_sdk/storage/struct.Storage.html#method.instance)
+
+Storage here is done for a small amount of persistent data tightly coupled with the contract instance. Data is loaded from the ledger every time the contract instance is loaded and it is limited by the ledger entry size, typically in the order of 100 KB serialized.
+
+#### Use cases
+
+- Small data directly associated with the contract, such as admin details, configuration settings, and tokens. For small, frequently used data that is integral to the contract instance and benefits from being loaded every time the contract is used.
+- Examples: Contract admin details, configuration settings, tokens operated by the contract.
+
+#### Cost
+
+- Likely cheaper than storing data separately in persistent storage.
+
+#### Lifetime
+
+- Similar lifetime properties to persistent storage but does not appear in the ledger footprint.
+
+## 5. **Contract size**
+
+The size of your smart contract's `wasm` binary influences the cost of running your smart contract on the Stellar network. Larger `wasm` binaries require more processing power and memory to execute, leading to higher gas costs. Larger binaries also cost more gas to deploy and invoke.
+
+To optimize the size of your `wasm` binary, you can:
+
+- Remove unnecessary code
+- Minimize the use of external dependencies
+- Use built-in tools to optimize the size of your `wasm` binary
+
+#### Example (Non-Optimized Contract) ❎
+
+```rust
+#![no_std]
+
+extern crate serde;
+extern crate serde_json;
+extern crate regex;
+
+use soroban_sdk::{contract, contractimpl, log, vec, Env, Symbol};
+use serde::{Deserialize, Serialize};
+use regex::Regex;
+
+#[contract]
+pub struct ExampleContract;
+
+#[contractimpl]
+impl ExampleContract {
+    // Function to log a message
+    pub fn log_message(env: Env, message: Symbol) {
+        log!(&env, "Message: {}", message);
+
+        // Unnecessary use of serde and regex
+        let _data: Vec<u8> = serde_json::to_vec(&message).unwrap();
+        let _re = Regex::new(r"\w+").unwrap();
+    }
+}
+```
+
+:::danger
+
+- Unnecessary Libraries: The contract includes serde, serde_json, and regex, which are not essential for the contract's primary functionality.
+- Increased WASM Size: Including these libraries adds unnecessary bloat, increasing the size of the compiled WASM file
+
+:::
+
+#### Example (Optimized Contract) ✅
+
+```rust
+#![no_std]
+use soroban_sdk::{contract, contractimpl, log, Env, Symbol};
+
+#[contract]
+pub struct ExampleContract;
+
+#[contractimpl]
+impl ExampleContract {
+    // Function to log a message
+    pub fn log_message(env: Env, message: Symbol) {
+        log!(&env, "Message: {}", message);
```

Comment:
This is a pretty contrived example. I would either get rid of this section or, ideally, come up with a better example where the unoptimized contract imports a third-party library that could be replaced with built-in methods.